app/services/legacy/dataset_import_service.rb

```diff
@@ -56,8 +56,10 @@ def create_datafiles(dataset)
       datafile = Link.find_or_create_by(url: legacy_datafile["url"], dataset_id: dataset.id)
       base_attributes = create_resource_base_attributes(legacy_datafile, dataset)
       date_attributes = create_datafile_date_attributes(legacy_datafile)
+      datafile.day = date_attributes[:day]
```

Comment:
Doesn't this do the same job?
```
datafile.assign_attributes(date_attributes)
```

Reply:
Assign attributes will save the `day` to the database. The idea is to leave that to the `before_save` call back so that the way dates are saved is consistent for datafiles that are imported and those made on publish